### PR TITLE
Better Optimizer and LR Scheduler kwargs specification + DenseTrainable

### PR DESCRIPTION
**Pull request by RussellALA**
- Added new settings to TrainableHParams:
  - betas: tuple/list of betas for Adam/RMSProp (momentum/RMS weight) 
  - name: Name of the experiment (for pytorch_lightning logging names)
- Fixed default LR scheduler (instead of None, which throws an error, it is now an empty scheduler, that does not decrease the learning rate)

### CODE REVIEW

File: src/lightning_trainable/trainable.py [hunk 1]
```diff
@@ -22,6 +22,7 @@ class TrainableHParams(HParams):
     max_epochs: int | None
     max_steps: int = -1
     optimizer: str = "adam"
+    betas: tuple | list = (0.9, 0.999)
```
Comment (fdraxler) [inline src/lightning_trainable/trainable.py:25]: I think it is good to be able to change this, but we should rather make optimizer a dict (optionally), and retrieve kwargs from that -- otherwise we end up adding parameters of all different kinds of optimizers to a huge single HParam

File: src/lightning_trainable/trainable.py [hunk 2]
```diff
@@ -31,6 +32,7 @@ class TrainableHParams(HParams):
     profiler: str | Profiler | None = None
     lr_scheduler: str | None = None
     num_workers: int = 4
+    name: str = "lightning_logs"
```
Comment (fdraxler) [inline src/lightning_trainable/trainable.py:35]: `log_dir` is currently a parameter to `__init__` as it does not affect the behavior of model or training. I think we should make it consistent.

File: src/lightning_trainable/trainable.py [hunk 3]
```diff
@@ -102,7 +105,11 @@ def configure_optimizers(self):
                     interval="step"
                 )
             case None:
-                lr_scheduler = None
+                lr_scheduler = dict(
```
Comment (fdraxler) [inline src/lightning_trainable/trainable.py:108]: I think it is cleaner to return `[optimizer], lr_schedulers` below and set `lr_schedulers = []` if none is specified

File: src/lightning_trainable/trainable.py (hunk 2 above)
Comment (fdraxler) [reply src/lightning_trainable/trainable.py:35]: Or maybe what did you want to achieve with `name` when you have `log_dir`?

File: src/lightning_trainable/trainable.py (hunk 1 above)
Comment (RussellALA) [reply src/lightning_trainable/trainable.py:25]: Makes sense. There are already optimizer parameters in the HParam (`weight_decay`, `learning_rate`). I suggest to move weight decay to the new kwargs dict. Learning rate as well or keep it explicit?

Comment (fdraxler) [reply src/lightning_trainable/trainable.py:25]: I don't mind `weight_decay` and `learning_rate` as all optimizers support them, but it is indeed cleaner to put them inside. But really, I don't care.

File: src/lightning_trainable/trainable.py (hunk 3 above)
Comment (LarsKue) [reply src/lightning_trainable/trainable.py:108]: We can also just return `optimizer` when no `lr_scheduler` is specified, and return a `dict` containing both otherwise.

File: src/lightning_trainable/trainable.py (hunk 1 above)
Comment (LarsKue) [reply src/lightning_trainable/trainable.py:25]: I am in favour of moving both into the `optimizer` dict, it is more consistent and not overly verbose. Still, I think using just `optimizer="adam"` should be allowed. Another concern: What do we call the optimizer name in the dict? `name` or `kind` come to mind, but `algorithm` is more technically accurate.
